pkg/services/ngalert/schedule/fetcher.go

```diff
@@ -43,6 +43,17 @@ func (sch *schedule) updateSchedulableAlertRules(ctx context.Context) error {
 			time.Since(start).Seconds())
 	}()
 
+	if !sch.schedulableAlertRules.isEmpty() {
+		keys, err := sch.ruleStore.GetAlertRulesKeysForScheduling(ctx)
+		if err != nil {
+			sch.log.Warn("failed to fetch alert rule keys.", "err", err)
+		}
+		if !sch.schedulableAlertRules.needsUpdate(keys) {
```

Comment:
If Line 47 returns an error then `needsUpdate` will return true?

Reply:
yes, that is my intention. If the auxiliary call failed we always try to do the main one. Kinda retry in case of db issues. I am not 100% certain about this approach, though